File: RedSheet.DbEntities/Models/EventPlanningAction.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("EventPlanningActions", Schema = "dbo")]
    [Serializable]
    public partial class EventPlanningAction
    {

        [Required]
        [MaxLength(50)]
        public string EventPlanningActionBy { get; set; }

        [Required]
        [MaxLength(400)]
        public string EventPlanningActionDetail { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int EventPlanningActionId { get; set; }

        [Required]
        public System.DateTime EventPlanningActionOn { get; set; }

        [RelationshipTableAttribue("ProjectPreparations", "dbo")]
        //Relationships
        public int ProjectPreparationId { get; set; }

        [ForeignKey("ProjectPreparationId")]
        public ProjectPreparation ProjectPreparation { get; set; }

        public EventPlanningAction()
        {
        }

    }
}

File: redsheet-portal/src/app/database-models/event-planning-action.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectPreparation,  vEventPlanningActionRecord  } from './'
export class EventPlanningAction {
    constructor(eventPlanningAction?: EventPlanningAction  | vEventPlanningActionRecord )  {
        let properties = [ "eventPlanningActionBy", "eventPlanningActionDetail", "eventPlanningActionId", "eventPlanningActionOn", "projectPreparationId",];
        for (let property of properties)
            if (eventPlanningAction && eventPlanningAction[property])
                this[property] = eventPlanningAction[property];
    }

    @required()
    @maxLength(50)
	eventPlanningActionBy : string =   undefined;

    @required()
    @maxLength(400)
	eventPlanningActionDetail : string =   undefined;

	eventPlanningActionId : number =   0 ;

    @required()
	eventPlanningActionOn : Date =   undefined;

    @range(0,2147483647)
	projectPreparationId : number =   undefined;
	projectPreparation : ProjectPreparation  ;


}
